runtime/runtime/src/lib.rs

```diff
@@ -2171,6 +2103,90 @@ struct ProcessReceiptsResult {
     processed_yield_timeouts: Vec<PromiseYieldTimeout>,
 }
 
+/// This struct is a convenient way to hold the processing state during [Runtime::apply].
+struct ApplyProcessingState<'a> {
```

Comment:
That's cool, and feel free to disregard my earlier comments about order of parameters :)